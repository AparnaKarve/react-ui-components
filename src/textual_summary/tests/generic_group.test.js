import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GenericGroup from '../generic_group';
import GenericTableRow from '../generic_table_row';

describe('GenericGroup', () => {
  it('renders title and rows', () => {
    const groupData = {
      title: 'Foo Bar',
      items: [{hoverClass: 'bar'}],
    };

    const wrapper = shallow(<GenericGroup items={groupData.items} title={groupData.title} onClick={e => null} />);
    expect(wrapper.html()).toContain(groupData.title);
    expect(wrapper.find(GenericTableRow).length).toEqual(1);
  });
});
