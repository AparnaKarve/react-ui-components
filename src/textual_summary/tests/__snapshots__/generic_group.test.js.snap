// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericGroup renders just fine 1`] = `
<table
  className="table table-bordered table-hover table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        align="left"
        colSpan="2"
      >
        Foo Bar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="bar"
      onClick={[Function]}
      title={undefined}
    >
      <td
        className="label"
      >
        foo
      </td>
      <td>
        
        <span>
           
          bar
        </span>
      </td>
    </tr>
    <tr
      className="bar"
      onClick={[Function]}
      title={undefined}
    >
      <td
        className="label"
      >
        mrkef?
      </td>
      <td>
        
        <span>
           
          v zime!
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
